src/include/cell.i

```diff
@@ -112,8 +115,15 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
             LF_SET(WT_CELL_TS_DURABLE_STOP);
         }
     }
-    if (tw->prepare)
+    if (LF_ISSET(
+          WT_CELL_TS_START | WT_CELL_TS_DURABLE_START | WT_CELL_TS_STOP | WT_CELL_TS_DURABLE_STOP))
+        r->page_with_ts = true;
```

Comment:
Maybe we should only set if not already true. Similarly other variables.

Reply:
I am not sure there is any performance benefit. We can either read a value from pointer `r` and then do a comparison, or, do a comparison on local variable and then store a value in member of pointer `r`.